### PR TITLE
Fix with latest arkworks version.

### PR DESCRIPTION
**Pull request by mmaker**

### CODE REVIEW

File: src/fields/fp/mod.rs [hunk 1]
```diff
@@ -424,8 +425,9 @@ impl<F: PrimeField> AllocatedFp<F> {
         // The high level logic is as follows:
         // We want to check that self - other != 0. We do this by checking that
         // (self - other).inverse() exists. In more detail, we check the following:
-        // If `should_enforce == true`, then we set `multiplier = (self - other).inverse()`,
-        // and check that (self - other) * multiplier == 1. (i.e., that the inverse exists)
+        // If `should_enforce == true`, then we set `multiplier = (self -
+        // other).inverse()`, and check that (self - other) * multiplier == 1.
+        // (i.e., that the inverse exists)
```
Comment (Pratyush) [inline src/fields/fp/mod.rs:430]:
```suggestion
        // If `should_enforce == true`, then we set `multiplier = (self - other).inverse()`, 
        // and check that (self - other) * multiplier == 1. (i.e., that the inverse exists)
```

File: src/fields/mod.rs [hunk 2]
```diff
@@ -20,8 +20,9 @@ pub mod quadratic_extension;
 /// That is, it implements the R1CS equivalent of `ark_ff::Fp*`.
 pub mod fp;
 
-/// This module contains a generic implementation of "nonnative" prime field variables.
-/// It emulates `Fp` arithmetic using `Fq` operations, where `p != q`.
+/// This module contains a generic implementation of "nonnative" prime field
+/// variables. It emulates `Fp` arithmetic using `Fq` operations, where `p !=
+/// q`.
```
Comment (Pratyush) [inline src/fields/mod.rs:25]:
```suggestion
/// variables. It emulates `Fp` arithmetic using `Fq` operations, where `p != q`.
```

File: src/fields/nonnative/allocated_field_var.rs [hunk 3]
```diff
@@ -620,9 +633,10 @@ impl<TargetField: PrimeField, BaseField: PrimeField>
         Ok(bits)
     }
 
-    /// Allocates a new non-native field witness with value given by the function `f`.  Enforces
-    /// that the field element has value in `[0, modulus)`, and returns the bits of its binary
-    /// representation. The bits are in little-endian (i.e., the bit at index 0 is the LSB) and the
+    /// Allocates a new non-native field witness with value given by the
+    /// function `f`.  Enforces that the field element has value in `[0,
+    /// modulus)`, and returns the bits of its binary representation. The
+    /// bits are in little-endian (i.e., the bit at index 0 is the LSB) and the
```
Comment (Pratyush) [inline src/fields/nonnative/allocated_field_var.rs:639]:
```suggestion
    /// function `f`.  Enforces that the field element has value in `[0, modulus)`,
    /// and returns the bits of its binary representation.
    /// The bits are in little-endian (i.e., the bit at index 0 is the LSB) and the
```

File: src/groups/curves/short_weierstrass/bls12/mod.rs [hunk 4]
```diff
@@ -42,8 +42,10 @@ impl<P: Bls12Parameters> G1PreparedVar<P> {
         let x = self.0.x.value()?;
         let y = self.0.y.value()?;
         let infinity = self.0.infinity.value()?;
-        let g = GroupAffine::new(x, y, infinity);
-        Ok(g.into())
+        let g = infinity
+            .then(|| GroupAffine::zero())
+            .unwrap_or_else(|| GroupAffine::new(x, y)).into();
```
Comment (Pratyush) [inline src/groups/curves/short_weierstrass/bls12/mod.rs:47]:
```suggestion
            .then_some(GroupAffine::zero())
            .unwrap_or( GroupAffine::new(x, y)).into();
```

File: src/groups/curves/short_weierstrass/mod.rs [hunk 5]
```diff
@@ -14,20 +16,22 @@ use crate::{fields::fp::FpVar, prelude::*, ToConstraintFieldGadget, Vec};
 pub mod bls12;
 
 /// This module provides a generic implementation of G1 and G2 for
-/// the [\[MNT4]\](<https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.20.8113&rep=rep1&type=pdf>)
+/// the [\[MNT4]\](<https://citeseerx.ist.psu.edu/viewdoc/downl&mut oad?doi=10.1.1.20.8113&rep=rep1&type=pdf>)
```
Comment (Pratyush) [inline src/groups/curves/short_weierstrass/mod.rs:19]:
```suggestion
/// the [\[MNT4]\](<https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.20.8113&rep=rep1&type=pdf>)
```

**Review by Pratyush — APPROVED** [review]

**Comment by mmaker** [comment]
Squashed together! tests are still failing.

**Pull request merged**